### DISCUSSION TITLE
22 marzo 2022 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#8] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/0a9cc5fc-0669-4214-a815-6322fa68410f)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/2e08708d-0fba-4ab1-9337-a4bf844da9d9)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/8e70025f-5023-444c-b798-48ca7cc1d36b)

**Comment by niccolo-fato** [comment]
![2024-01-13 11 37 14](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/a64bb2a3-3005-4cf7-8ad5-2cdb5e4bda88)

**Comment by Elia-Belli** [comment]
**(1)**
- Partecipazioni al progetto Supercar: $Part\\_ SC= Partecipazione\bowtie_{(CodicePROG=Codice)}(\sigma_{(Titolo\ =\ SUPERCAR)}Progetto)$
- Settori in cui lavorano degli Impiegati del Progetto Supercar: $Settore\\_ SC=\pi_{(Settore)}(Impiegato\bowtie_{(Codice=CodiceImp)}Part\\_ SC)$
- Settore Sviluppo Supercar: $Set\\_ Sviluppo\\_ SC =\sigma_{(Funzione\ =\ Sviluppo)}(Settore\\_ SC \bowtie_{(Settore\ =\ Codice)}Settore)$
- Risposta: $\pi_{(CF,Nome,Cognome)}(Impiegato\bowtie_{(Codice\ =\ Caposettore)}Set\\_ Sviluppo)$

**(2)** 
- Partecipazione al progetto CasaMia: $Part\\_ CasaMia = (\sigma_{(Titolo\ =\ CASAMIA)}Progetto)\bowtie_{(Codice\ =\ CodicePROG)}(Partecipazione)$
- Impiegati nel progetto CasaMia: $Imp\\_ CasaMia= \pi_{(CF,Nome,Cognome,DataInizio)}(Impiegato\bowtie_{(Codice=CodiceIMP)}Part\\_CasaMia)$
- Copia Impiegati: $Imp\\_ Copia=\rho_{(CFC\leftarrow CF,NomeC\leftarrow Nome,CognomeC\leftarrow Cognome,DataInizioC\leftarrow DataInizio)}Imp\\_ CasaMia$
- Impiegati che hanno partecipato almeno prima di un altro: $No\\_ Imp=\pi_{(CF,Nome,Cognome)}(\sigma_{(DataInizio \lt DataInizioC)}(Imp\\_ CasaMia\times Imp\\_ Copia))$
> Non è possibile il calcolo diretto, infatti se selezioniamo $DataInizio\gt DataInizioC$ prenderemmo gli impiegati che hanno iniziato a lavorare dopo **almeno** un altro impiegato, mentre noi vogliamo che abbia iniziato dopo **tutti** gli altri
- Risposta: $(\pi_{(CF,Nome,Cognome)}(Imp\\_ CasaMia))- No\\_ Imp$
